kazoo/recipe/lock.py

```diff
@@ -82,18 +83,31 @@ class Lock(object):
     # sequence number. Involved in read/write locks.
     _EXCLUDE_NAMES = ["__lock__"]
 
-    def __init__(self, client, path, identifier=None):
+    def __init__(
+        self, client, path, identifier=None, additional_lock_patterns=()
+    ):
         """Create a Kazoo lock.
 
         :param client: A :class:`~kazoo.client.KazooClient` instance.
         :param path: The lock path to use.
-        :param identifier: Name to use for this lock contender. This
-                           can be useful for querying to see who the
-                           current lock contenders are.
-
+        :param identifier: Name to use for this lock contender. This can be
+                          useful for querying to see who the current lock
+                          contenders are.
+        :param additional_lock_patterns: Strings that will be used to
+                                         identify other znode in the path
+                                         that should be considered contenders
+                                         for this lock.
+                                         Use this for cross-implementation
+                                         compatibility.
+
+        .. versionadded:: 2.7.1
+            The additional_lock_patterns option.
         """
         self.client = client
         self.path = path
+        self._exclude_names = set(
+            self._EXCLUDE_NAMES + list(additional_lock_patterns)
```

Comment:
This is the "meat":  Add any additional patterns to the list of recognized contenders patterns

Reply:
Thank you for this